Source/Core/DolphinQt/ToolBar.cpp

```diff
@@ -14,14 +14,28 @@
 #include "DolphinQt/Settings.h"
 #include "DolphinQt/ToolBar.h"
 
-static QSize ICON_SIZE(32, 32);
+static const QSize ICON_SIZE(32, 32);
+static const QSize ICON_SIZE_SMALL(16, 16);
 
 ToolBar::ToolBar(QWidget* parent) : QToolBar(parent)
 {
-  setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
+  auto update_compact_view_lambda = [this] {
+    if (Settings::Instance().IsCompactViewEnabled())
```

Comment:
This can be rewritten as:
```c++
setIconSize(compact ? ICON_SIZE_SMALL : ICON_SIZE);
setToolButtonStyle(compact ? Qt::ToolButtonIconOnly : Qt::ToolButtonTextUnderIcon);
``` 

(Making use of the ``CompactViewChanged`` ``bool compact``parameter)